tests/test_core_configure.py

```diff
@@ -129,6 +132,174 @@ def test_verify_spec_succeeds(capsys):
     core_configure.verify_spec(spec)
 
 
+def test_verify_spec_fails_multiple_roles_own_schema(capsys):
+    """
+    Should successfuly detect config that defines two roles as an owner of a single schema.
+    """
+    spec_yaml = """
+    jfinance:
+        is_superuser: yes
+        owns:
+            schemas:
+                - finance_documents
+    jfauxnance:
+        is_superuser: yes
+        owns:
+            schemas:
+                - finance_documents
+    """
+    spec = yaml.load(spec_yaml)
+    with pytest.raises(SystemExit):
+        core_configure.verify_spec(spec)
+
+    out, _ = capsys.readouterr()
+    expected = core_configure.MULTIPLE_SCHEMA_OWNER_ERR_MSG.format('finance_documents', 'jfinance, jfauxnance')
+    assert expected in out
+
+def test_verify_spec_fails_multiple_roles_own_schema_personal_schema(capsys):
+    """
+    Should successfuly detect config that defines a role with invalid access to a personal schema.
+    """
+    spec_yaml = """
+    jfinance:
+        is_superuser: yes
+        has_personal_schema: yes
+        owns:
+            schemas:
+                - finance_documents
+    jfauxnance:
+        is_superuser: yes
+        owns:
+            schemas:
+                - jfinance
+    """
+    spec = yaml.load(spec_yaml)
+    with pytest.raises(SystemExit):
+        core_configure.verify_spec(spec)
+
+    out, _ = capsys.readouterr()
+    expected = core_configure.MULTIPLE_SCHEMA_OWNER_ERR_MSG.format('jfinance', 'jfinance, jfauxnance')
+    assert expected in out
+
+
+def test_verify_spec_fails_role_defined_multiple_times(tmpdir, capsys):
+    """
+    Verify should successfully detect config that defines two roles as an owner of a single schema.
+    """
+    spec_path = tmpdir.join('spec.yml')
+    spec_path.write("""
+    jfinance:
+        is_superuser: yes
+        owns:
+            schemas:
+                - finance_documents
+    jfinance:
+        is_superuser: yes
+        owns:
+            schemas:
+                - even_more_finance_documents
+    patty:
+        is_superuser: yes
+        owns:
+            schemas:
+                - tupperwear
+    """)
+    with pytest.raises(SystemExit):
+        core_configure.load_spec(spec_path.strpath)
+
+    out, _ = capsys.readouterr()
+    expected = core_configure.DUPLICATE_ROLE_DEFINITIONS_ERR_MSG.format('jfinance')
+    assert expected in out
+
+
+def test_verify_spec_fails_object_referenced_read_write_schemas(capsys):
```

Comment:
You can combine this and the two following tests using a parametrized test, e.g.:
```
@pytest.mark.parametrize('objkind', ['schemas', 'tables', 'sequences'])
def test_verify_spec_fails_object_referenced_read_write(capsys, objkind):
    spec_yaml = """
    margerie:
        can_login: true
        privileges:
            {0}:
                read:
                    - big_bad
        [...]
        .format(objkind)
    [...]
```